### DISCUSSION TITLE
Graphql and Nextjs

### DISCUSSION THREAD

**Discussion by Joel-Ajayi** [vercel/next.js#25608] [Help]
I want to build a full graphql server in Nextjs, is this possible ?
And can I deploy to vercel or heroku with the same domain name, while the server is in nextjs ?

**Comment by Cerchie** [comment]
@Joel-Ajayi ayi it certainly is! 
Can you give me more detail on what you're thinking of building?

I'm an engineer at [StepZen](https://stepzen.com/), and we have a [Next.js sample project](https://github.com/stepzen-samples/netlify-next-app) here deployable to Netlify or Vercel, and the graphql part is done for you with our StepZen  GraphQL API endpoint. It might help 
you out! 

If you have any questions about the sample or StepZen, or even just GraphQL, feel free to reach out, either in this conversation  or on our [discord server](https://discord.com/channels/768229795544170506/768229795544170509)
